core/include/scipp/core/variable.h

```diff
@@ -419,6 +419,12 @@ class SCIPP_CORE_EXPORT Variable {
   Variable &operator|=(const Variable &other) &;
   Variable &operator|=(const VariableConstProxy &other) &;
 
+  Variable &operator&=(const Variable &other) &;
```

Comment:
We actually do not need this overload since `Variable` is implicitly convertible to `VariableConstProxy`. I have been meaning to clean this up in some other cases (I would not recommend doing so in this PR).